src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/CaCaseloadService.kt

```diff
@@ -0,0 +1,492 @@
+        val deliusRecord = deliusRecords.find { d -> d.otherIds.nomsNumber == prisoner.prisonerNumber }
+        if (deliusRecord != null) {
+          return@map ManagedCase(
+            nomisRecord = prisoner,
+            cvlFields = cvl,
+            deliusRecord = DeliusRecord(
+              deliusRecord,
+              ManagedOffenderCrn(
+                staff = deliusRecord.offenderManagers.find { om -> om.active }?.staffDetail,
+              ),
+            ),
+          )
+        }
+        return@map ManagedCase(nomisRecord = prisoner, cvlFields = cvl)
+      }
+  }
+
+  private fun mapCasesToComs(casesToMap: List<CaCase>): List<CaCase> {
+    val cases = splitCasesByComDetails(casesToMap)
+
+    val noComPrisonerNumbers = cases.withNoComId.map { c -> c.prisonerNumber }
+    val deliusRecords = probationSearchApiClient.searchForPeopleByNomsNumber(noComPrisonerNumbers)
+
+    var caCaseList: List<CaCase> = mutableListOf()
+
+    // if no code or username, hit delius to find COM details
+    caCaseList += cases.withNoComId.map { caCase ->
+
+      val com =
+        deliusRecords.find { d -> d.otherIds.nomsNumber == caCase.prisonerNumber }?.offenderManagers?.find { om -> om.active }?.staffDetail
+      if (com != null && !com.unallocated!!) {
+        return@map caCase.copy(
+          probationPractitioner = ProbationPractitioner(
+            staffCode = com.code,
+            name = com.let { "${it.forenames} ${it.surname}" },
+          ),
+        )
+      }
+      return@map caCase
+    }
+
+    // If COM username but no code, do a separate call to use the data in CVL if it exists. Should help highlight any desync between Delius and CVL
+    val comUsernames = cases.withStaffUsername.map { c -> c.probationPractitioner?.staffUsername!! }
+    val coms = communityApiClient.getStaffDetailsByUsername(comUsernames)
+
+    caCaseList += cases.withStaffUsername.map { caCase ->
+      val com =
+        coms.find { com -> com.username?.lowercase() == caCase.probationPractitioner?.staffUsername?.lowercase() }
+      if (com != null) {
+        return@map caCase.copy(
+          probationPractitioner = ProbationPractitioner(
+            staffCode = com.staffCode,
+            name = com.staff?.let { "${it.forenames} ${it.surname}" },
+          ),
+        )
+      }
+      return@map caCase
+    }
+
+    // If already have COM code and name, no extra calls required
+    caCaseList += cases.withStaffCode.map { caCase ->
+      return@map caCase.copy(
+        probationPractitioner = ProbationPractitioner(
+          staffCode = caCase.probationPractitioner?.staffCode,
+          name = caCase.probationPractitioner?.name,
+        ),
+      )
+    }
+
+    return caCaseList
+  }
+
+  private fun buildCaseload(cases: List<CaCase>, searchString: String?, view: String): CaCaseLoad {
+    val showAttentionNeededTab = cases.any { it.tabType == CaViewCasesTab.ATTENTION_NEEDED }
+    val searchResults = applySearch(searchString, cases)
```

Comment:
Ya, it's always better to move most of the possible functionality to backend.
So, it will be easy in the future if we want to make search functionality as common in the backend.